queue/driver_mongo.go

```diff
@@ -677,24 +677,40 @@ func (d *mongoDriver) GetAllAttempts(ctx context.Context, id string) ([]amboy.Jo
 }
 
 func (d *mongoDriver) Put(ctx context.Context, j amboy.Job) error {
-	ji, err := registry.MakeJobInterchange(j, d.opts.Format)
-	if err != nil {
-		return errors.Wrap(err, "converting in-memory job to interchange job")
+	return d.PutMany(ctx, []amboy.Job{j})
+}
+
+// PutMany enqueues multiple jobs on the queue.
+// Returns an error if any of the jobs cannot be inserted. If a non-duplicate-job error
+// is encountered it is returned. If every job is a duplicate job error then if any
+// of them is a duplicate scope error a duplicate scope error is returned.
+// If none of them is a duplicate scope error then a duplicate job error
+// is returned.
+func (d *mongoDriver) PutMany(ctx context.Context, jobs []amboy.Job) error {
+	var jobInterchanges []any
+	for _, j := range jobs {
+		ji, err := registry.MakeJobInterchange(j, d.opts.Format)
+		if err != nil {
+			return errors.Wrap(err, "converting in-memory job to interchange job")
+		}
+		ji.Scopes = j.EnqueueScopes()
+		d.addMetadata(ji)
+		jobInterchanges = append(jobInterchanges, ji)
 	}
 
-	ji.Scopes = j.EnqueueScopes()
-
-	d.addMetadata(ji)
-
-	if _, err = d.getCollection().InsertOne(ctx, ji); err != nil {
+	if _, err := d.getCollection().InsertMany(ctx, jobInterchanges, options.InsertMany().SetOrdered(false)); err != nil {
+		jobIDs := make([]string, 0, len(jobs))
+		for _, j := range jobs {
+			jobIDs = append(jobIDs, j.ID())
```

Comment:
Instead of listing all the job IDs here, would it be more useful to return the [set difference](https://github.com/evergreen-ci/utility/blob/main/slice.go#L88-L108) with the result's [InsertedIDs](https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo#InsertManyResult) slice? Since we are continuing on error, some IDs may have been inserted.

Reply:
done